src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -20,26 +20,51 @@
 
 using System;
 using System.ComponentModel.Composition;
+using System.Threading;
 using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.IssueVisualization.Editor;
+using SonarLint.VisualStudio.IssueVisualization.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList;
+using SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableDataSource;
 using SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Contract;
 using SonarQube.Client;
+using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Api
 {
     [Export(typeof(IOpenInIDERequestHandler))]
     internal class OpenInIDERequestHandler : IOpenInIDERequestHandler
     {
+        private readonly IToolWindowService toolWindowService;
         private readonly IOpenInIDEStateValidator ideStateValidator;
         private readonly ISonarQubeService sonarQubeService;
+        private readonly IHotspotToIssueVisualizationConverter converter;
+        private readonly ILocationNavigator navigator;
+        private readonly IHotspotsStore hotspotsStore;
+        private readonly IOpenInIDEFailureInfoBar failureInfoBar;
         private readonly ILogger logger;
 
         [ImportingConstructor]
-        public OpenInIDERequestHandler(IOpenInIDEStateValidator ideStateValidator, ISonarQubeService sonarQubeService, ILogger logger)
+        public OpenInIDERequestHandler(
+            IToolWindowService toolWindowService,
+            IOpenInIDEStateValidator ideStateValidator,
+            ISonarQubeService sonarQubeService,
+            IHotspotToIssueVisualizationConverter converter,
+            ILocationNavigator navigator,
+            IHotspotsStore hotspotsStore,
+            IOpenInIDEFailureInfoBar failueInfoBar,
+            ILogger logger)
```

Comment:
This component uses a lot of other services. Fortunately, the processing is relatively simple:
* always clear the previous info bar, if there was one
* always switch to the hotspots tool window
* stop processing the request in the event of any failure and show a gold bar